enumerable.rb

```diff
@@ -0,0 +1,349 @@
+      end
+      if self.size == 0
+        true
+      elsif word.class == Regexp
+
+        for element in self
+          if element.match(word)
+            j += 1
+          end
+        end
+        j == self.length ? true : false
+
+      elsif word == Numeric
+
+        for element in self
+          if element.class <= word
+            j += 1
+          end
+        end
+        j == self.length ? true : false
+      elsif word != nil
+        i = 0
+        while i < self.size
+          if self[i - 1] == self[i]
+            k += 1
+          end
+          i += 1
+        end
+
+        for element in self
+          if element.class == word
+            j += 1
+          end
+        end
+        j == self.length ? true : false
+        k == self.length ? true : false
+
+      else
+        for element in self do
+          if element == false or element == nil
+            k += 1
+          end
+        end
+        k == 0 ? true : false
+      end
+    else
+      i = 0
+      j = 0
+      while i < self.size
+        yield self[i]
+        i += 1
+      end
+
+      for element in self
+
+        if yield(element) == false or yield(element) == nil
+          j += 1
+        end
+      end
+      j == 0 ? true : false
+
+    end
+  end
+
+  def my_any?(word = nil)
+    unless block_given?
+      i = 0
+      j = 0
+      while i < self.size
+        self[i]
+        i += 1
+      end
+      if self.size == 0
+        false
+      elsif word.class == Regexp
+
+        for element in self
+          if element.match(word)
+            j += 1
+          end
+        end
+        j > 0 and j <= self.length ? true : false
+
+      elsif word != nil
+        i = 0
+        k = 0
+        while i < self.size
+          if self[i - 1] == self[i]
+            k += 1
+          end
+          i += 1
+        end
+
+        for element in self
+          if element == word
+            k += 1
+          end
+        end
+        k > 0 ? true : false
+
+      elsif word == Numeric
+        for element in self
+          if element.class <= word
+            j += 1
+          end
+        end
+
+        j > 0 and j <= self.length ? true : false
+
+      else
+        k = 0
+        for element in self do
+          if element == false or element == nil
+            k += 1
+          end
+        end
+        k != self.size ? true : false
+
+      end
+
+    else
+      i = 0
+      j = 0
+      while i < self.size
+        yield self[i]
+        i += 1
+      end
+
+      for element in self
+        if yield element
+          j += 1
+        end
+      end
+      j > 0 and j <= self.size ? true : false
+    end
+  end
+
+  def my_none?(word = nil)
+    unless block_given?
+      i = 0
+      j = 0
+      while i < self.size
+        self[i]
+        i += 1
+      end
+      if self.size == 0
+        true
+      elsif word.class == Regexp
+
+        for element in self
+          if element.match(word)
+            j += 1
+          end
```

Comment:
* [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5,6,7].my_none?(String) != [1,2,3,4,5,6,7].none?(String)
```